### DISCUSSION TITLE
Need some help with sizing and responsiveness

### DISCUSSION THREAD

**Discussion by IsabelBou** [chakra-ui/chakra-ui#8159] [Q&A]
Hello! This is my first project as Front-end using Chakra-UI (And I am loving it so far!) and I'm having a few issues that apparently shouldn't be happening. Every time I patch an issue something else happens, and I worry I might not be doing things as they should be done due to lack of understanding; so before I go further into my project and increase its complexity I would like to make sure its base is well done. 

I would appreciate validation of things being done the way they should be, or some guidance on how it should be corrected or improved (so I can learn how it should be done next time), or if it may be a bug. I have been reading the official docs and related issues here and on stackoverflow, but my understanding may be wrong and I would appreciate feedback from people who have more experience than me and solid knowledge. Thank you for your help and patience in advance!

Mainly my concerns have to do with sizing and responsiveness. Cards should by default extend towards both ends of the screen (if I understood the docs correctly), and not only do mine not do that, but they used to shrink as much as possible until recently. I created a component that bundles accordions and a MIDI player inside a card, but I can't get them to be shown according to the browser's size most of the time. 

I currently have a simple grid with the custom cards, since I intend to have two of them side by side when the screen is wide enough, and have only one otherwise. The change upon breakpoint works perfectly! But it only looks good on a small range of screen widths.

![gif showcasing the app's "responsiveness"](https://github.com/chakra-ui/chakra-ui/assets/67393544/7d86a555-2a72-4095-90e0-c93762bd67a7)

## Wide Screen

When the screen gets bigger, the content gets divided in two columns (good!), but it also stops adapting to the browser's width. I believed this was a card issue, since they stay the same size from there, but I think it's more like the cards have no space to grow to, since the root element has a very wide margin. I've tried modifying it with several style props from the main App function and not only does it never look entirely good, but I also worry modifying the core config theme might break something else (I've read about CSS reset as a possible solution, for example).

![A wider browser causes the #root div to have very wide margins](https://github.com/chakra-ui/chakra-ui/assets/67393544/952d1a0b-e404-4224-9821-f4b207c87c2e)

### Previous and maybe related issues?

In fact, initially the background would only span like half the height, and this was solved recently by encapsulating the content of `ChakraProvider` inside a `Container` with `minHeight="100vh"`, but this caused a permanent vertical scrollbar on the right even when there's no content below.

![blank bottom without minHeight](https://github.com/chakra-ui/chakra-ui/assets/67393544/22acb29f-f888-4f0e-911e-ac4c5d4b0486)

Similarly, this `Container` solved a previous issue where the composite cards would shrink as much as possible, slightly growing when the accordion opened. This happened by adding to it `width="100vw"`.

![very small composite card, accordion closed](https://github.com/chakra-ui/chakra-ui/assets/67393544/74f6e222-0694-4d19-b427-323880915375)

![still small composite card, accordion expanded](https://github.com/chakra-ui/chakra-ui/assets/67393544/3602f250-c1bd-4845-8a0a-a31efdcf657b)

But I wonder if, instead of adding the `Container` inside `ChakraProvider`, I shouldn't be configuring something else instead, or if this might be bad practice or cause issues in the future. I couldn't come up with anything better and couldn't find other solutions or root causes, either.
 
## Narrow Screen
 
When the screen gets smaller only one column is displayed (again, good!), and for a small length the margins on the sides of the pages will adapt, too, but then a horizontal scroll appears and pushes all the content to the right upon width change; not only that, but if you scroll to the right, the part that was hidden is empty (no background). The card remains the same size until there's around twice the page padding's width hidden behind the scrollbar, **and then** it starts shrinking.

![narrow browser causes the page to scroll to the right](https://github.com/chakra-ui/chakra-ui/assets/67393544/ffc7f9f1-816b-40f6-a670-5fc91b28f28d)

![scrolling to the right reveals an empty background, although it still considered part of the root's content and padding](https://github.com/chakra-ui/chakra-ui/assets/67393544/697969ef-a6ee-4396-ac1b-fef067a49fe9)

This is where I have no idea what I should even be doing or why it is happening :_)

## Context

I am using Vite 5.0.0, Chakra 2.8.2 and ReactJS 18.2.0. At first I started my project with Create React App until I found out it was deprecated due to it causing a lot of problems, then created a new Vite.js project and moved my code there (piece by piece, in order to make sure everything worked fine).

The cards that don't have Play/Pause buttons inside are regular cards I added just to see if they would act the same way as the composite ones, which have an accordion and a button.

The content inside the composite cards is a Verovio generated SVG (which scales to fit its container) and a button that toggles a MIDI file generated out of Verovio, which is synthesized with JZZ.js, but I don't think these are the cause of the problems so I won't include their code here and instead will try to include just the code I believe may be causing them. If more context is needed, I'll be more than happy to provide it! The full code can be checked on [my repository](https://github.com/IsabelBou/Musischolar)

This is my main function App():

```javascript
function App() {
  return (
    <ChakraProvider theme = {customTheme} >
      <Container minHeight = "100vh" width = "100vw">  
          <Header />
          <br></br>
          <Card> Test </Card>
          <br></br>
          <SimpleGrid 
            spacing = {4} 
            columns={{base: 1, sm: 1, md: 2}}> {/* If screen about the size of a phone (portrait), 1 column. Otherwise, 2 of them*/}
            <CustomCard {...cards.TEMPLATE}/>
            <CustomCard {...cards.TEMPLATE}/>
            <Card> Test </Card>
          </SimpleGrid>
      </Container>
    </ChakraProvider>
  )
}
``` 

My composite card looks like this:

```javascript
const customCard = (props) => {
    const { title, description, score } = props;
    return (
        <Card>
            <CardHeader>
                <Accordion allowToggle allowMultiple>
                    <AccordionItem border='none'>                  
                        <AccordionButton >
                            <Box flex='1' textAlign='center'>
                                {title}
                            </Box>
                            <AccordionIcon />
                        </AccordionButton>
                        <AccordionPanel>
                            {description}
                        </AccordionPanel>
                    </AccordionItem>
                </Accordion>
            </CardHeader>
            <CardBody>
                <Player score = { score }/>
            </CardBody>
        </Card>
    )
}

export default customCard;
``` 

The theme index:

```javascript
const customTheme = extendTheme(
  {
    styles: {
      global: (props) => ({
        "html, body": {
          background: mode("primary.50", "primary.950")(props),
        },
      }),
    },
    colors: themeColors,
      fonts: {
          body: "Verdana, sans-serif",
          heading: "Georgia, serif",
          mono: "Menlo, monospace",
      },
      components: {
        Card: cardTheme,
        Accordion: cardAccordion,
        Tooltip: tooltipTheme,
      },
  },
  withDefaultColorScheme({ colorScheme: 'primary' }),
  {
    config: {
      disableTransitionOnChange: false
    },
  }
)

export default customTheme
``` 

the card's theme:

```javascript
const { definePartsStyle, defineMultiStyleConfig } =
  createMultiStyleConfigHelpers(cardAnatomy.keys)

const baseStyle = definePartsStyle({
  container: {
    backgroundColor: "primary.75",
    _dark:{
      backgroundColor: "primary.780",
    }
  },
  header: {
    paddingBottom: '0.5rem',
  },
  body: {
    paddingTop: '0.25rem',
    paddingBottom: '1rem'
  },
  footer: {
    paddingTop: '0.5rem',
  },
})

export const cardTheme = defineMultiStyleConfig({ baseStyle })
``` 

the accordion's theme (contained inside cards):

```javascript
const { definePartsStyle, defineMultiStyleConfig } =
  createMultiStyleConfigHelpers(accordionAnatomy.keys)

const baseStyle = definePartsStyle({
  button: {
    allowToggle: true,
    allowMultiple: true,
    fontSize: '1.5rem',
  },
  panel: {
    paddingBottom: '0.25rem',
    fontSize: '0.75rem',
  },
})

export const cardAccordion = defineMultiStyleConfig({ baseStyle })
``` 

##  TL;DR

My questions are:

-  If cards are meant to always extend to the max screen width, why have mine never done that? Did I overwrite something while customizing their style? According to the docs the `Container` didn't seem necessary for cards to stretch.
-  How can I get the margins from the `#root` div to stop taking over most of the browser's width? Is this the cause of the cards' not growing, or is that due to something else entirely?
- Is there anything I did that should be done in any other way? I get the feeling I've been patching things up when problems appeared instead of dealing with the real issue underneath, and worry it might cause more problems in the future. Also I'm just getting started, so every feedback I can get from people with more experience will be very valuable to me!

Any help or advice will be much appreciated! Thank you for reading this far!
